File: questao_09/questao_09.go
package main

import "fmt"

type Imprimivel interface {
	Imprimir()
}

type Relatorio struct {
	conteudo string
}

func (r Relatorio) Imprimir() {
	fmt.Println("Imprimindo Relatório:", r.conteudo)
}

type Contrato struct {
	partes string
}

func (c Contrato) Imprimir() {
	fmt.Println("Imprimindo Contrato:", c.partes)
}

func main() {
	var imprimiveis []Imprimivel
	imprimiveis = append(imprimiveis, Relatorio{"Relatório Mensal"}, Contrato{"Contrato de Prestação de Serviços"})

	for _, i := range imprimiveis {
		i.Imprimir()
	}
}

File: questao_09/questao_09.py
from abc import ABC, abstractmethod

class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self):
        pass

class Relatorio(Imprimivel):
    def __init__(self, conteudo):
        self.conteudo = conteudo

    def imprimir(self):
        print(f"Relatório: {self.conteudo}")

class Contrato(Imprimivel):
    def __init__(self, partes):
        self.partes = partes

    def imprimir(self):
        print(f"Contrato: {self.partes}")

relatorio = Relatorio("Relatório Mensal")
contrato = Contrato("Contrato de Prestação de Serviços")

relatorio.imprimir()
contrato.imprimir()
